services/src/main/java/org/keycloak/authentication/authenticators/client/JWTClientValidator.java

```diff
@@ -192,12 +194,50 @@ public void validateToken() {
         }
     }
 
+    private long calculateLifespanInSeconds()  {
+        if ((token.getExp() == null || token.getExp() <= 0) && (token.getIat() == null || token.getIat() <= 0)) {
+            throw new RuntimeException("Token cannot be validated. Neither the exp nor the iat claim are present.");
+        }
+
+        if (token.getExp() == null || token.getExp() <= 0) {
+            // exp not present but iat present, just allow a short period of time from iat (10s)
+            // The JWT MUST contain an "exp" by spec??? https://datatracker.ietf.org/doc/html/draft-jones-oauth-jwt-bearer-03#section-3
```

Comment:
The spec says that `exp` is compulsory, but null is allowed as it was accepted before.

Reply:
In the last RFC both `iat` and `exp` are marked as required. So I have changed this comment and linked the last RFC.